### PR TITLE
Add JNI bindings for extractAllRecord

### PR DESCRIPTION
**Pull request by anthony-chang**
Part of https://github.com/rapidsai/cudf/issues/11033

Adds `extractAllRecord` to the JNI layer. For now this only supports `idx` 0 using `cudf::strings::findall_record`. 

Corresponding spark plugin PR: https://github.com/NVIDIA/spark-rapids/pull/5947

### CODE REVIEW

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

File: java/src/test/java/ai/rapids/cudf/ColumnVectorTest.java [hunk 1]
```diff
@@ -4097,17 +4097,34 @@ void testStringFindOperations() {
...
+  void testExtractRe() {
+      try (ColumnVector input = ColumnVector.fromStrings("a1", "b2", "c3", null);
+            Table expected = new Table.TestBuilder()
+                    .column("a", "b", null, null)
+                    .column("1", "2", null, null)
+                    .build();
+            Table found = input.extractRe("([ab])(\\d)")) {
+          assertTablesAreEqual(expected, found);
+      }
+  }
```
Comment (anthony-chang) [inline java/src/test/java/ai/rapids/cudf/ColumnVectorTest.java:4110]: Indentation change

File: java/src/main/java/ai/rapids/cudf/ColumnView.java [hunk 2]
```diff
@@ -3238,6 +3238,25 @@ public final Table extractRe(String pattern) throws CudfException {
     return new Table(extractRe(this.getNativeView(), pattern));
   }
 
+  /**
+   * Extracts all strings that match the given regular expression and corresponds to the 
+   * regular expression group index. Any null inputs also result in null output entries.
+   * 
+   * For supported regex patterns refer to:
+   * @link https://docs.rapids.ai/api/libcudf/nightly/md_regex.html
+   * @param pattern the regex pattern
```
Comment (ttnghia) [inline java/src/main/java/ai/rapids/cudf/ColumnView.java:3247]:
```suggestion

   * @param pattern the regex pattern
```

File: java/src/main/java/ai/rapids/cudf/ColumnView.java [hunk 3]
```diff
@@ -3926,6 +3945,15 @@ private static native long stringReplaceWithBackrefs(long columnView, String pat
    */
   private static native long[] extractRe(long cudfViewHandle, String pattern) throws CudfException;
 
+  /**
+   * Native method for extracting all results corresponding to group idx from a regular expression.
+   * @param nativeHandle native handle of the cudf::column_view being operated on.
```
Comment (ttnghia) [inline java/src/main/java/ai/rapids/cudf/ColumnView.java:3950]:
```suggestion

   * @param nativeHandle native handle of the cudf::column_view being operated on.
```

File: java/src/main/java/ai/rapids/cudf/ColumnView.java [hunk 4]
```diff
@@ -3238,6 +3238,25 @@ public final Table extractRe(String pattern) throws CudfException {
...
 
+  /**
+   * Extracts all strings that match the given regular expression and corresponds to the 
+   * regular expression group index. Any null inputs also result in null output entries.
+   * 
+   * For supported regex patterns refer to:
+   * @link https://docs.rapids.ai/api/libcudf/nightly/md_regex.html
+   * @param pattern the regex pattern
+   * @param idx the regex group index
+   * @return a new column vector of extracted matches
```
Comment (ttnghia) [inline java/src/main/java/ai/rapids/cudf/ColumnView.java:3249]:
```suggestion
   * For supported regex patterns refer to:
   * @link https://docs.rapids.ai/api/libcudf/nightly/md_regex.html
   
   * @param pattern The regex pattern
   * @param idx The regex group index
   * @return A new column vector of extracted matches
```

File: java/src/main/java/ai/rapids/cudf/ColumnView.java [hunk 5]
```diff
@@ -3926,6 +3945,15 @@ private static native long stringReplaceWithBackrefs(long columnView, String pat
    */
   private static native long[] extractRe(long cudfViewHandle, String pattern) throws CudfException;
 
+  /**
+   * Native method for extracting all results corresponding to group idx from a regular expression.
+   * @param nativeHandle native handle of the cudf::column_view being operated on.
+   * @param pattern string regex pattern.
+   * @param idx regex group index. An idx of 0 means matching the entire regex.
+   * @return native handle of a string column of the result.
```
Comment (ttnghia) [inline java/src/main/java/ai/rapids/cudf/ColumnView.java:3953]:
```suggestion
   * Native method for extracting all results corresponding to group idx from a regular expression.
   *
   * @param nativeHandle Native handle of the cudf::column_view being operated on.
   * @param pattern String regex pattern.
   * @param idx Regex group index. A 0 value means matching the entire regex.
   * @return Native handle of a string column of the result.
```

File: java/src/main/java/ai/rapids/cudf/ColumnView.java [hunk 6]
```diff
@@ -3238,6 +3238,26 @@ public final Table extractRe(String pattern) throws CudfException {
...
 
+  /**
+   * Extracts all strings that match the given regular expression and corresponds to the 
+   * regular expression group index. Any null inputs also result in null output entries.
+   * 
+   * For supported regex patterns refer to:
+   * @link https://docs.rapids.ai/api/libcudf/nightly/md_regex.html
+   
+   * @param pattern The regex pattern
+   * @param idx The regex group index
```
Comment (revans2) [inline java/src/main/java/ai/rapids/cudf/ColumnView.java:3249]: Can we document that only 0 is currently supported?

**Review by revans2 — APPROVED** [review]
Just a documentation nit.

**Review by revans2 — APPROVED** [review]

**Comment by raydouglass** [comment]
okay to test

**Pull request merged**